### DISCUSSION TITLE
Why is there no route that allows an array of products to be added to the cart?

### DISCUSSION THREAD

**Discussion by multiplehats** [woocommerce/woocommerce-gutenberg-products-block#9775] [Q&A - Extension Integrations and Customizations] [answered]
Hi all,

I was looking at the Store API docs, and it seems like (or at least to me it does), that the [`/wc/store/v1/cart/items` ](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/src/StoreApi/docs/cart-items.md#add-cart-item)doesn't allow an array of products to be added.

I've also traced it back here: https://github.com/woocommerce/woocommerce-blocks/blob/cbc55eff55ed442ef8f99e1dc2ea22b6222dfe9f/src/StoreApi/Routes/V1/CartItems.php#L98

This confused me at first because the route path implies you can add multiple items.

Maybe I'm missing something? If not, I would love to know what the preferred way is of adding say, 10 items to the cart.

**Comment by opr** [comment, marked as answer]
Hi @multiplehats - please take a look at this document

https://github.com/woocommerce/woocommerce-blocks/blob/trunk/src/StoreApi/docs/cart.md#add-item

Specifically the "Batching" section of the Add Item route.

You can add multiple items by making a batch request containing multiple `cart/add-item` requests.

The JSON payload for adding multiple items would look like this:

```json
{
	"requests": [
		{
			"path": "/wc/store/v1/cart/add-item",
			"method": "POST",
			"cache": "no-store",
			"body": {
				"id": 26,
				"quantity": 1
			},
			"headers": {
				"Nonce": "1db1d13784"
			}
		},
		{
			"path": "/wc/store/v1/cart/add-item",
			"method": "POST",
			"cache": "no-store",
			"body": {
				"id": 27,
				"quantity": 1
			},
			"headers": {
				"Nonce": "1db1d13784"
			}
		}
	]
}
```

I'll close this discussion since it has a definitive answer, but please feel free to reopen it if you would like to discuss further!

**Reply by multiplehats** [reply]
Ohhhhh, I totally missed that.  Thanks a lot!

If you want I can add a PR with a link to that doc in case others are running into the same issue.

**Reply by opr** [reply]
No worries, it was in a different document to the one you were looking at, actually. A PR would be awesome! Feel free to assign me as a reviewer and I'll take a look today or tomorrow.

**Reply by multiplehats** [reply]
Appreciate it. Unfortunately I can't tag/assign you as a reviewer, but just added the PR: https://github.com/woocommerce/woocommerce-blocks/pull/9776
